File: LeetCode/Java/Easy/BinaryTreePaths.java
package Easy;

import java.util.*;

public class BinaryTreePaths {

	public void main(String[] args) {

	}
	
	public List<String> binaryTreePaths(TreeNode root){
		List<String> answer = new ArrayList<String>();
		if (root != null){
			searchTree(root, "", answer);
		}
		return answer;
	}
	
	public void searchTree(TreeNode root, String path, List<String> answer){
		if (root.left == null && root.right == null){
			answer.add(path + root.val);
		}
		if (root.left != null){
			searchTree(root.left, path + root.val + "->", answer);
		}
		if (root.right != null){
			searchTree(root.right, path + root.val + "->", answer);
		}
	}

}

File: LeetCode/Python/Easy/BinaryTreePaths.py
class Solution(object):
    def binaryTreePaths(self, root):
        result = []
        if root is None:
            return result

        self.binaryTreePathHelper(root, result, '')
        return result

    def binaryTreePathHelper(self, root, result, curr):
        if root.left is None and root.right is None:
            result.append(curr + str(root.val))
        else:
            curr = curr + str(root.val) + '->'
            if root.left is not None:
                self.binaryTreePathHelper(root.left, result, curr)
            if root.right is not None:
                self.binaryTreePathHelper(root.right, result, curr)
